### PR TITLE
chore: add script to extract front matter key from markdown files

### PR DESCRIPTION
**Pull request by seidior**
## Description

Adding in a script to extract a given key from Markdown frontmatter, then added additional guidance for Python unit testing in the Readme.

### Approach

You'll need to set up a virtual environment for Python. Follow the steps in the revised Readme.

### Steps to Test

To test, run `yarn test:python`.

## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: content/scripts/test_get_frontmatter_key.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
```
Comment (adidner) [inline content/scripts/test_get_frontmatter_key.py:2]: [dust] I get that this is the expected output, but I still found it a bit confusing before looking at the tests, any way we could try and make this a tad clearer?
image

**Review by adidner — APPROVED** [review]
Seems good, minor question and comment

**Review by AnJuHyppolite — COMMENTED** [review]
Looks good save for Aaron's comments!

File: README.md [hunk 2]
```diff
@@ -80,10 +80,19 @@ Before you can run locally, you will need to:
   `./web/.env-template` file.
 - create a `./api/.env` that includes all the values laid out in the
   `./api/.env-template` file.
+- create a `.venv` virtual environment and install requirements if working on
+  Python:
+
+```shell
+python3 -m venv .venv
+source .venv/bin/activate
+pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (seidior) [reply README.md:89]: You would need to `source .venv/bin/activate` each time, as you would need to do that _before_ running `yarn test:python` unfortunately, as it relies on bash/zsh functionality in the same process.

I've added VSCode functionality that automatically detects the virtual environment and sources it when you open a new terminal. Is there a way I can do the same for your editor as well?

File: content/scripts/test_get_frontmatter_key.py (hunk 1 above)
Comment (seidior) [reply content/scripts/test_get_frontmatter_key.py:2]: Yeah... I didn't love that either, much as I don't love our `console.log()` outputs when we run Jest. I've addressed this in a revised commit that buffers (`-b`) the output, so it shouldn't display anymore. Thanks for the feedback :)

**Pull request merged**